src/pfs/pfs.go

```diff
@@ -78,3 +78,32 @@ func (b *Branch) NewCommit(id string) *Commit {
 func (b *Branch) String() string {
 	return b.Repo.String() + "@" + b.Name
 }
+
+func NewListCommitRequest(repo *Repo) *ListCommitRequest {
+	return &ListCommitRequest{Repo: repo}
+}
+
+func (lcr *ListCommitRequest) WithTo(to *Commit) *ListCommitRequest {
```

Comment:
This API is a little misleading with the pointers.  The requests aren't immutable, they are being modified and then returned.  I would expect the receiver not to be a pointer, but maybe there are performance issues with copying like that.

Reply:
Yeah, I had concerns about the performance, but those may be unfounded. I would like to use pointers because that is what gRPC takes, and we could change this to a copy / clone under the hood later.

Reply:
I have a feeling for almost all requests it'd be fine to clone. It maybe gets to be a bit of overhead if it's heavily chained. That being said I don't find the api that confusing, with does imply that it's not mutating anything but I can't really think of a better name, you could maybe do `SetTo` but that somehow doesn't seem as good to me.